### PR TITLE
snapstate: add support for UC20 gadget in {Config,Gadget}Defaults

### PR DESCRIPTION
**Pull request by mvo5**
Right now we don't set the constraints.SystemSeed property when
parsing the gadget.yaml from snapstate. This breaks seeding on
a UC20 system when it reaches the gadget default stage.

Tests are missing and we probably need a check in image.go as well
so that we detect if someone tries to build a UC20 image with an
incompatible gadget early. But I have many meetings today so may not
get to it today :(

### CODE REVIEW

**Comment by pedronis** [comment]
We probably want to check for this much earlier,  in image and I think actually in seed itself.

**Comment by pedronis** [comment]
Probably around here:

https://github.com/snapcore/snapd/blob/master/seed/seed20.go#L404

**Comment by pedronis** [comment]
We probably need to add a gadget.ReadInfoFromSnapFile(snap.Container, constraints) (on top of InfoFromGadgetYaml)

**Comment by mvo5** [comment]
We should care about volumes in:
- image
- seeding
- upgrades of gadgets
But not do this everytime we load the gadget.

**Comment by xnox** [comment]
I wonder if I should learn how to build snapd snaps with PRs and self inject them into an image.

Not sure if I can actually do this, as dangerous options are not implemented for ubuntu-image/prepare-image yet i don't think.

File: overlord/snapstate/snapstate.go [hunk 1]
```diff
@@ -2374,11 +2374,8 @@ func ConfigDefaults(st *state.State, deviceCtx DeviceContext, snapName string) (
 		return nil, state.ErrNoState
 	}
 
-	constraints := &gadget.ModelConstraints{
-		Classic: release.OnClassic,
-	}
-
-	gadgetInfo, err := gadget.ReadInfo(info.MountDir(), constraints)
+	// at this point we don't need constraints checks, they happend earlier
```
Comment (zyga) [inline overlord/snapstate/snapstate.go]: There's a typo here:
```suggestion
	// at this point we don't need constraints checks, they happened earlier
```

**Comment by mvo5** [comment]
This is a bit of a kitchen sink right now - I will pull out bits and propose separately.

**Pull request closed without merging**